File: app/src/main/java/com/jakting/opengapps/MainActivity.java
package com.jakting.opengapps;

import android.content.Context;
import android.content.res.Configuration;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getDarkModeStatus(this)) {
            setTheme(R.style.RnDark);
            setBarIconColor(true);
        } else {
            setTheme(R.style.RnLight);
            setBarIconColor(false);
        }
        setContentView(R.layout.activity_main);
        BottomNavigationView navView = findViewById(R.id.nav_view);
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_device, R.id.navigation_download, R.id.navigation_about)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(navView, navController);

    }

    //检查当前系统是否已开启暗黑模式
    public static boolean getDarkModeStatus(Context context) {
        int mode = context.getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
        return mode == Configuration.UI_MODE_NIGHT_YES;
    }

    //白底黑字，黑底白字
    public void setBarIconColor(boolean dark_open) {
        //白底黑字
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            this.getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            if (dark_open) {
                this.getWindow().setStatusBarColor(getColor(R.color.colorPrimaryDarkDark));
                this.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN & ~View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
            } else {
                this.getWindow().setStatusBarColor(getColor(R.color.colorPrimaryDark));
                this.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
            }
        }
    }

}

File: app/src/main/java/com/jakting/opengapps/MainActivity.kt
package com.jakting.opengapps

import android.content.Context
import android.content.res.Configuration
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.jakting.opengapps.ui.device.DeviceFragment
import com.jakting.opengapps.ui.download.DownloadFragment
import com.jakting.opengapps.ui.misc.MiscFragment
import com.jakting.opengapps.utils.logd
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), ViewPager.OnPageChangeListener {

    private val deviceFragment: DeviceFragment = DeviceFragment()
    private val downloadFragment: DownloadFragment = DownloadFragment()
    private val miscFragment: MiscFragment = MiscFragment()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!getDarkModeStatus(this)){
            window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
        }
        setContentView(R.layout.activity_main)
        viewPaper.addOnPageChangeListener(this)
        nav_view.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        viewPaper.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
            override fun getItem(position: Int): Fragment {
                when (position) {
                    0 -> {
                        logd("deviceFragment")
                        return deviceFragment
                    }
                    1 -> {
                        logd("downloadFragment")
                        return downloadFragment
                    }
                    2 -> {
                        logd("miscFragment")
                        return miscFragment
                    }
                    else -> {
                        return deviceFragment
                    }
                }

            }

            override fun getCount(): Int {
                return 3
            }
        }

    }

    override fun onBackPressed() {
        finish()
    }

    private val mOnNavigationItemSelectedListener: BottomNavigationView.OnNavigationItemSelectedListener =
        object : BottomNavigationView.OnNavigationItemSelectedListener {
            override fun onNavigationItemSelected(item: MenuItem): Boolean {
                //点击BottomNavigationView的Item项，切换ViewPager页面
//menu/navigation.xml里加的android:orderInCategory属性就是下面item.getOrder()取的值
                viewPaper.currentItem = item.order
                return true
            }
        }

    override fun onPageScrolled(
        position: Int,
        positionOffset: Float,
        positionOffsetPixels: Int
    ) {
    }

    override fun onPageSelected(position: Int) { //页面滑动的时候，改变BottomNavigationView的Item高亮
        nav_view.getMenu().getItem(position).isChecked = true
    }

    override fun onPageScrollStateChanged(state: Int) {}

    fun getDarkModeStatus(context: Context): Boolean {
        val mode: Int =
            context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        return mode == Configuration.UI_MODE_NIGHT_YES
    }
}
